packages/teleport/src/services/mfa/makeMfaDevice.ts

```diff
@@ -0,0 +1,25 @@
+import moment from 'moment';
+import { displayDate } from 'shared/services/loc';
+import { MfaDevice } from './types';
+
+export default function makeMfaDevice(json): MfaDevice {
+  const { id, name, lastUsed, addedAt } = json;
+
+  let type: MfaDevice['type'];
+  if (json.type === 'TOTP') {
+    type = 'Authenticator App';
+  }
+  if (json.type === 'U2F') {
+    type = 'Hardware Key';
+  }
+
+  return {
+    id,
+    name,
+    type,
+    registeredDate: moment(addedAt).unix(),
+    lastUsedDate: moment(lastUsed).unix(),
```

Comment:
hm, is this conversion required?

Reply:
Yes. The `addedAt` and `lastUsed` returned from the backend are formatted strings eg. `"2021-08-18T18:39:23.522007Z"`. Converting them to unix timestamps makes sorting possible.

Reply:
@rudream no need to use `unix` nor `moment` here, just create `Date` objects as follows
 
```
  registeredDate: new Date(addedAt),
  lastUsedDate: new Date(lastUsed),
```